### PR TITLE
Chore/add tests and optimizations

### PR DESCRIPTION
**Pull request by AStox**

### CODE REVIEW

File: src/Connector.sol [hunk 1]
```diff
@@ -132,7 +133,7 @@ contract CentrifugeConnector {
         uint128 price
     ) public onlyRouter {
         Tranche storage tranche = tranches[poolId][trancheId];
-        require(tranche.latestPrice > 0, "CentrifugeConnector/invalid-pool-or-tranche");
+        require(tranche.lastPriceUpdate > 0, "CentrifugeConnector/invalid-pool-or-tranche");
```
[the remark this answers is not in the document]
Comment (AStox) [reply src/Connector.sol:136]: Although it's very unlikely, token price could someday be 0, and then this check which should just be trying to see if the tranche exists or not, would throw an error. But since we set lastPriceUpdate on tranche creation now, a lastPriceUpdate of 0 will *always* mean the tranche doesn't exist.

File: src/Connector.sol [hunk 2]
```diff
@@ -157,8 +158,9 @@ contract CentrifugeConnector {
         uint256 amount
     ) public onlyRouter {
         RestrictedTokenLike token = RestrictedTokenLike(tranches[poolId][trancheId].token);
-        require(address(token) != address(0), "CentrifugeConnector/unknown-token");
-        require(token.hasMember(user), "CentrifugeConnector/not-a-member");
+        // // require(address(token) != address(0), "CentrifugeConnector/unknown-token");
```
[the remark this answers is not in the document]
Comment (AStox) [reply src/Connector.sol]: Fixed up!

File: src/Connector.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (NunoAlexandre) [reply src/Connector.sol:136]: Very good point 💯

**Review by NunoAlexandre — APPROVED** [review]
LGTM but would like to have @offerijns's look on this as well

File: src/token/memberlist.sol [hunk 3]
```diff
@@ -28,7 +28,7 @@ contract Memberlist {
     }
 
     function updateMember(address usr, uint validUntil) public auth {
-        require((safeAdd(block.timestamp, minimumDelay)) < validUntil, "invalid-validUntil");
+        require((safeAdd(block.timestamp, minimumDelay)) < validUntil, "cent/invalid-validUntil");
```
Comment (offerijns) [inline src/token/memberlist.sol:31]: Why this change?

**Review by offerijns — COMMENTED** [review]
Just two questions

For the error naming, I think we should just try to move to Solidity 0.8 soon and then we can use typed errors.

File: src/token/memberlist.sol (hunk 3 above)
Comment (NunoAlexandre) [reply src/token/memberlist.sol:31]: yeh, in fact a lot of other errors are prefixed with `CentrifugeConnector/`

Comment (AStox) [reply src/token/memberlist.sol]: I just removed this change to keep these contracts unaltered

File: src/token/erc20.sol [hunk 4]
```diff
@@ -72,7 +72,7 @@ contract ERC20 {
         emit Transfer(src, dst, wad);
         return true;
     }
-    function mint(address usr, uint wad) external virtual auth {
+    function mint(address usr, uint wad) public virtual auth {
```
[the remark this answers is not in the document]
Comment (AStox) [reply src/token/erc20.sol]: Ah my bad. This was left over from the mint change.

**Review by offerijns — APPROVED** [review]
